build/Targets/Imports.targets

```diff
@@ -62,9 +62,11 @@
       <PropertyGroup>
         <AllowedReferenceRelatedFileExtensions>$(AllowedReferenceRelatedFileExtensions);.pdb</AllowedReferenceRelatedFileExtensions>
         <CopyNuGetImplementations>true</CopyNuGetImplementations>
+        <OutDir Condition="'$(TargetFrameworkIdentifier)' == '.NETPortable'">$(OutDir)CoreClrTest</OutDir>
+        <OutDir Condition="'$(TargetFrameworkIdentifier)' != '.NETPortable'">$(OutDir)UnitTests\Portable\</OutDir>
       </PropertyGroup>
     </When>
-    <When Condition="'$(RoslynProjectType)' == 'Custom' OR $(RoslynProjectType) == 'DeploymentTest'">
```

Comment:
I believe this cleans up all uses of `DeploymentTest` in our projects.  Can you remove the corresponding checking code in BuildBoss as well? 

https://github.com/dotnet/roslyn/blob/d81ed64ed9e437c118b95288358ff9ae683b66c6/src/Tools/BuildBoss/ProjectCheckerUtil.cs